Feature/scroll view2

gggangmin opened the pull request:
Shop PCT의 기존 ScrollViewContent를 ScrollViewContent_class로 변경하고 ScrollViewContent_item을 추가함
ScrollViewContent_class의 버튼 눌렀을 때 해당 버튼의 id를 ScrollViewContent_item로 전송하며 화면 출력
기존의 ScrollViewContent_class는 비활성화

gggangmin committed:
Feat: Scroll View for item on class

+ 기존 scrollViewContent-> scrollViewContent_class로 변경
+ scrlooViewContent_item 추가
+ button 눌렀을 때 새로운 scrollview나타나도록 추가

VanSubstance commented:
Merge 대상 branch를 dev로 바꿔주세요.

VanSubstance commented:
정확히 무엇을 의도한 기능인지 설명이 필요합니다.
어떤 목적으로 해당 기능을 만드신건지 여쭤봐도 될까요 ?

VanSubstance commented on ProjectFaker/Assets/Scripts/UI/ButtonManager.cs:
```diff
@@ -7,18 +7,35 @@ public class ButtonManager : MonoBehaviour
 
     [SerializeField]
-    private Transform scrollViewContent;
+    private Transform scrollViewContent_class;
+    [SerializeField]
+    private Transform scrollViewContent_item;
```
private 변수 명은 camelCase를 사용하는 것을 원칙으로 합니다 .
`scrollViewContentClass`
`scrollViewContentItem`

VanSubstance commented on ProjectFaker/Assets/Scripts/UI/ButtonManager.cs:
```diff
@@ -7,18 +7,35 @@ public class ButtonManager : MonoBehaviour
     private Transform buttonTemplate;
 
-    public int btnCount = 10;
+    public int classBtnCount = 4;
+    public int itemBtnCount=5;
```
public 변수 명은 CapitalCase를 사용하는 것을 원칙으로 합니다.
`ClassBtnCount = 4;`
`ItemBtnCount = 5;`

VanSubstance commented on ProjectFaker/Assets/Scripts/UI/ButtonManager.cs:
```diff
@@ -7,18 +7,35 @@ public class ButtonManager : MonoBehaviour
+    public int classBtnCount = 4;
+    public int itemBtnCount=5;
     
     void Start()
```
접근자는 반드시 명시합니다.
`private`, `protected`, `public`

VanSubstance commented:
https://github.com/VanSubstance/faker/pull/23
이전 PR의 리뷰가 제대로 반영되지 않았습니다.
이전 리뷰에서 짚었던 부분 최대한 제외하고 이번 코드 리뷰 진행하겠습니다. 양쪽 다 수정 부탁드립니다.

VanSubstance commented on ProjectFaker/Assets/Scripts/UI/ButtonManager.cs:
```diff
@@ -7,18 +7,35 @@ public class ButtonManager : MonoBehaviour
+      ActiveScrollViewContent_class();
+    }
+
+    void ActiveScrollViewContent_class()
```
**접근자 명시**

VanSubstance replied:
Unity의 함수명은 `CapitalCase`를 원칙으로 합니다.

VanSubstance commented on ProjectFaker/Assets/Scripts/UI/ButtonManager.cs:
```diff
@@ -7,18 +7,35 @@ public class ButtonManager : MonoBehaviour
+      }
+    }
+
+    void ActiveScrollViewContent_item(int classId)
```
**접근자 명시**

VanSubstance replied:
Unity의 함수명은 `CapitalCase`를 원칙으로 합니다.

VanSubstance commented on ProjectFaker/Assets/Scripts/UI/ButtonManager.cs:
```diff
@@ -7,18 +7,35 @@ public class ButtonManager : MonoBehaviour
+    [SerializeField]
+    private Transform scrollViewContent_item;
     [SerializeField]
     private Transform buttonTemplate;
```
현재 코드에서 `buttonTemplate` 오브젝트의 UnityEngine.UI.Button 컴포넌트만 불러와서 사용하는 것으로 보입니다.
그렇다면 `Transform` 보다 해당 타입인 `UnityEngine.UI.Button`으로 선언해서 사용하는 것이 좋습니다.

VanSubstance commented on ProjectFaker/Assets/Scripts/UI/ButtonManager.cs:
```diff
@@ -7,18 +7,35 @@ public class ButtonManager : MonoBehaviour
+    void ActiveScrollViewContent_class()
+    {
+      for (int i = 0; i < classBtnCount; i++){
+        Transform button = Instantiate(buttonTemplate,scrollViewContent_class.transform);
```
`scrollViewContent_class`가 이미 `Transform`인 것으로 보입니다.
`~~.transform`을 호출할 필요가 없습니다.

VanSubstance reviewed (changes requested):
이전 PR에서 남겼던 리뷰가 제대로 반영되지 않은 것이 보입니다.
이전 리뷰와 이번 리뷰 전부 반영 부탁드립니다.

gggangmin committed:
Feat: Fix Scroll View

+ 변수명 camelCase, CapitalCase 반영
+ 함수명 CaplitalCase 반영
+ 접근자 명시
+ buttonTemplate 오브젝트 Transform->UnityEngine.UI.Button으로 변경
+ scrollViewClass transform 호출 제거

gggangmin commented:
> 이전 PR에서 남겼던 리뷰가 제대로 반영되지 않은 것이 보입니다. 이전 리뷰와 이번 리뷰 전부 반영 부탁드립니다.

확인되면 머지하겠습니다.

VanSubstance reviewed (approved):
lgtm :ant:
